rebar.config.script

```diff
@@ -0,0 +1,31 @@
+%% -*- mode: erlang; -*-
+case erlang:function_exported(rebar3, main, 1) of
+    true ->
+        %% rebar3
+        CONFIG;
+    false ->
+        %% rebar 2.x or older
+        %% rebuild deps
+        Rebar3Deps = proplists:get_value(deps, CONFIG),
+        Rebar3TestProf = proplists:get_value(test, proplists:get_value(profiles, CONFIG)),
+        {value, {deps, Rebar3TestDeps}, Rebar3TestProf2} = lists:keytake(deps, 1, Rebar3TestProf),
+
+        %% whenever adding a new Hex package dependency, this fun should be
+        %% updated
+        HexToRepo = fun(proper) -> "https://github.com/manopapad/proper.git";
+                       (covertool) -> "https://github.com/idubrov/covertool.git"
+                    end,
+
+        ConvertDep =
+            fun({DepName, {git, Repo, {ref, Tag}}}) ->
+                    {DepName, ".*", {git, Repo, {tag, Tag}}};
+               ({DepName, {git, Repo, {branch, Branch}}}) ->
+                    {DepName, ".*", {git, Repo, {branch, Branch}}};
+               ({HexDepName, HexDepVersion}) ->
```

Comment:
The objection, I assume, is to `({HexDepName, HexDepVersion})`?

I'd rather keep it because it's copy/paste from other projects, and if `proper` gets into Hex in a non-weird way (right now I think the Git and Hex releases don't match) then I'd want to swap the Hex link in.